internal/oci/runtime_oci.go

```diff
@@ -953,14 +957,23 @@ func (r *runtimeOCI) StopLoopForContainer(c *Container, bm kwait.BackoffManager)
 		}
 	}
 killContainer:
+	// TODO: Remove this function when we have standardized exit codes or
+	// error reporting from container runtimes. Current approach is a
+	// temporary workaround based on error string matching.
+	isProcessGoneError := func(err error) bool {
+		return errors.Is(err, ErrNoSuchProcess) ||
+			errors.Is(err, ErrNotRunning) ||
+			errors.Is(err, ErrInvalidProcess)
+	}
 	// We cannot use ExponentialBackoff() here as its stop conditions are not flexible enough.
 	kwait.BackoffUntil(func() {
-		if _, err := r.runtimeCmd("kill", c.ID(), "KILL"); err != nil {
+		if _, err := r.runtimeCmd("kill", c.ID(), "KILL"); err != nil && !isProcessGoneError(err) {
```

Comment:
@sohankunkerkar, do you plan to update `runtimeCmd()` to do a string match and return the proper error type? 

Otherwise, the `errors.Is()` wouldn't do us any good, as it does not do any string match; there is nothing that calls `<T>.Error()` and does a comparison of any sort with the string value there. An example:

```go
package main

import (
	"errors"
	"fmt"
)

var e = errors.New("test error")
var x = errors.New("test error")

func t1() error {
	return errors.New("test error")
}

func t2() error {
	return e
}

func t3() error {
	return errors.New("this is a test error of sorts")
}

func t4() error {
	return x
}

func main() {
	fmt.Println(errors.Is(t1(), e))
	fmt.Println(errors.Is(t2(), e))
	fmt.Println(errors.Is(t3(), e))
	fmt.Println(errors.Is(t4(), e))
}
```

```console
false
true
false
false
```

Internally, the error string is backed by a pointer to the internal `errors.errorString` type, which holds the actual text. This ensures that no two errors (types) with the same string are to be considered identical.

Reply:
Yeah, that's a fair point. Let me update the code to reflect this.

Reply:
@kwilczynski PTAL

Reply:
@sohankunkerkar, I left a few comments/ideas.